boring/src/asn1.rs

```diff
@@ -623,4 +633,15 @@ mod tests {
             .map(|object| object.to_string())
             .expect_err("parsing invalid OID should fail");
     }
+
+    #[test]
+    fn object_to_text() {
+        let oid = "2.16.840.1.101.3.4.2.1";
+        let object = Asn1Object::from_str(oid).unwrap();
+        assert_eq!(
+            object.to_text(false).unwrap(),
+            Nid::SHA256.long_name().unwrap()
+        );
+        assert_eq!(object.to_text(true).unwrap(), oid.to_string());
```

Comment:
API design comment: It's not at all obvious what the boolean parameter is doing at the call site. What do you think about using an enum instead?

Reply:
Agreed! This definitely needs some thought, and should have put a note that this was still wet code. I think our main options are:

1. stay close to the base API, using the non-obvious `bool`
2. use an `enum`
3. have an “opinionated” API that foregoes the option altogether (or has two separate methods with clearer names)

I don’t have much experience with other ffi-heavy crates (and so maybe there is a broader commonality that I‘m missing), but it seems like this crate is fairly opinionated, and so I might lean toward something like `name()` and `numerical_name()`

Reply:
That's true, looking at the rest of the crate for examples of similar functions-with-options sounds good.

Reply:
I’m not seeing any solid precedent, so I‘m going to go with a single new method that meets our needs:

- `oid_string`, which will call the FFI method with `no_name = 1`

When we go upstream, we can discuss about a second method (`to_text`?) with `no_name = 0`, or a single method with `enum` parameter.